internal/report/report.go

```diff
@@ -0,0 +1,183 @@
+const (
+	StatusPass  = "pass"
+	StatusFail  = "fail"
+	StatusWarn  = "warn"
+	StatusError = "error"
+	StatusSkip  = "skip"
+)
+
+func NewCPR(name string) *ClusterPolicyReport {
+	return &ClusterPolicyReport{
+		ClusterPolicyReport: v1alpha2.ClusterPolicyReport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              PrefixNameClusterPolicyReport + name,
+				CreationTimestamp: metav1.Now(),
+			},
+			Summary: v1alpha2.PolicyReportSummary{
+				Pass:  0, // count of policies with requirements met
+				Fail:  0, // count of policies with requirements not met
+				Warn:  0, // not used for now
+				Error: 0, // count of policies that couldn't be evaluated
+				Skip:  0, // count of policies that were not selected for evaluation
+			},
+			Results: []*v1alpha2.PolicyReportResult{},
+		},
+	}
+}
+
+func NewPR(namespace *v1.Namespace) *PolicyReport {
+	return &PolicyReport{
+		PolicyReport: v1alpha2.PolicyReport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              PrefixNamePolicyReport + namespace.Name,
+				Namespace:         namespace.Name,
+				CreationTimestamp: metav1.Now(),
+			},
+			Scope: &v1.ObjectReference{
+				Kind:            namespace.Kind,
+				Namespace:       "",
+				Name:            namespace.Name,
+				UID:             namespace.UID,
+				APIVersion:      namespace.APIVersion,
+				ResourceVersion: namespace.ResourceVersion,
+			},
+			Summary: v1alpha2.PolicyReportSummary{
+				Pass:  0, // count of policies with requirements met
+				Fail:  0, // count of policies with requirements not met
+				Warn:  0, // not used for now
+				Error: 0, // count of policies that couldn't be evaluated
+				Skip:  0, // count of policies that were not selected for evaluation
+			},
+			Results: []*v1alpha2.PolicyReportResult{},
+		},
+	}
+}
+
+func (r *ClusterPolicyReport) GetType() ResourceType {
+	return ClusterPolicyReportType
+}
+
+func (r *PolicyReport) GetType() ResourceType {
+	return PolicyReportType
+}
+
+func (r *PolicyReport) AddResult(policy policiesv1.Policy, resource unstructured.Unstructured, auditResponse *admv1.AdmissionReview, responseErr error) {
+	result := newResult(policy, resource, auditResponse, responseErr)
+	switch result.Result {
+	case StatusFail:
+		r.Summary.Fail++
+	case StatusError:
+		r.Summary.Error++
+	case StatusPass:
+		r.Summary.Pass++
+	}
+	r.Results = append(r.Results, result)
+}
+
+func (r *ClusterPolicyReport) AddResult(policy policiesv1.Policy, resource unstructured.Unstructured, auditResponse *admv1.AdmissionReview, responseErr error) {
+	result := newResult(policy, resource, auditResponse, responseErr)
+	switch result.Result {
+	case StatusFail:
+		r.Summary.Fail++
+	case StatusError:
+		r.Summary.Error++
+	case StatusPass:
+		r.Summary.Pass++
+	}
+	r.Results = append(r.Results, result)
+}
+
+func newResult(policy policiesv1.Policy, resource unstructured.Unstructured, auditResponse *admv1.AdmissionReview, responseErr error) *v1alpha2.PolicyReportResult {
```

Comment:
nitpicking: `newPolicyReportResult`

Reply:
Done in https://github.com/kubewarden/audit-scanner/pull/40/commits/898d19140dc59093af5165a216f4595643d2e004